Sources/Fluent/Database/Transactable.swift

```diff
@@ -11,12 +11,12 @@ extension Database: Transactable {
     /// be rolled back.
     ///
     /// note: the underlying driver must support transactions
-    public func transaction(_ closure: (Connection) throws -> ()) throws {
+    public func transaction<R>(_ closure: (Connection) throws -> R) throws -> R {
```

Comment:
do we need to add `@discardableResult` here to avoid getting warnings about unused results?